framework/src/media/MediaPlayerImpl.cpp

```diff
@@ -586,10 +586,18 @@ void MediaPlayerImpl::notifyObserver(player_observer_command_t cmd, ...)
 
 void MediaPlayerImpl::playback()
 {
-	ssize_t num_read = mInputHandler.read(mBuffer, (int)mBufSize);
+	ssize_t num_read = 0;
+	auto source = mInputHandler.getInputDataSource();
+	int channels = source->getChannels();
+	
+	if (channels == AUDIO_OUTPUT_CHANNELS) {
+		num_read = mInputHandler.read(mBuffer, (int)mBufSize);
+	} else {
+		num_read = mInputHandler.read(mBuffer, (int)mBufSize / AUDIO_OUTPUT_CHANNELS);
```

Comment:
This is out of point but it is not good that access another module's constant directly.